app/controllers/sales_controller.rb

```diff
@@ -35,8 +35,7 @@ def newShop
 
 
   def index
-  session[:current_sales_person_id]=1
-  @sale=Sale.find(session[:current_sales_person_id])
+  @sale=current_user
```

Comment:
You could make before_action callback in application controller which will assign variables. So in every controller, you don't have to define this.